core/src/main/java/org/bitcoinj/core/Address.java

```diff
@@ -71,6 +71,30 @@ private Address(NetworkParameters params, boolean p2sh, byte[] hash160) throws A
         this.p2sh = p2sh;
     }
 
+    /**
+     * Construct an address from its textual form.
+     *
+     * @param params
+     *            the expected network this address is valid for, or null if the network should be derived from the
+     *            textual form
+     * @param str
+     *            the textual form of the address, such as "17kzeh4N8g49GFvdDzSf8PjaPfyoD1MndL" or
+     *            "bc1qw508d6qejxtdg4y5r3zarvary0c5xw7kv8f3t4"
```

Comment:
how much do you dashify stuff like comments?

Reply:
In past, I wasn't, but at the end of this PR I will.